lms/services/auto_grading.py

```diff
@@ -63,7 +63,7 @@ def get_last_grades(
             select(GradingSyncGrade)
             .distinct(LMSUser.h_userid)
             .join(GradingSync)
-            .join(LMSUser)
+            .join(LMSUser, GradingSyncGrade.lms_user_id == LMSUser.id)
```

Comment:
This is the fix, adding the join condition explicitly, otherwise SQLA issues a join on GradingSync.created_by